base/asyncmap.jl

```diff
@@ -19,44 +19,97 @@ type AsyncCollector
     f
     results
     enumerator::Enumerate
-    ntasks::Int
+    max_tasks::Function
+    c::Channel{Tuple{Int, Any}}     # to communicate with the tasks
```

Comment:
a slightly more descriptive field name than `c` might be good for clarity's sake